File: src/data/rewards.ts
export interface Reward {
  id: string
  titleKey: string
  pts: number
}

export const rewards: Reward[] = [
  { id: 'extra5', titleKey: 'reward.extra5', pts: 300 },
  { id: 'credit5', titleKey: 'reward.credit5', pts: 800 },
  { id: 'freemonth', titleKey: 'reward.freemonth', pts: 2500 },
]

export interface Tier {
  key: 'bronze' | 'silver' | 'gold' | 'platinum'
  min: number
}

export const TIERS: Tier[] = [
  { key: 'bronze', min: 0 },
  { key: 'silver', min: 1500 },
  { key: 'gold', min: 5000 },
  { key: 'platinum', min: 12000 },
]

export const TIER_PERKS: Record<Tier['key'], string[]> = {
  bronze: ['tier.perk.earn', 'tier.perk.birthday'],
  silver: ['tier.perk.allBronze', 'tier.perk.bonus5', 'tier.perk.priority'],
  gold: ['tier.perk.allSilver', 'tier.perk.bonus10', 'tier.perk.freeDelivery', 'tier.perk.early'],
  platinum: ['tier.perk.allGold', 'tier.perk.bonus15', 'tier.perk.concierge', 'tier.perk.giftAnnual'],
}

export function tierInfo(points: number): { current: Tier; next: Tier | null; progress: number } {
  let idx = 0
  for (let i = 0; i < TIERS.length; i++) if (points >= TIERS[i].min) idx = i
  const current = TIERS[idx]
  const next = TIERS[idx + 1] ?? null
  const progress = next ? Math.min(1, (points - current.min) / (next.min - current.min)) : 1
  return { current, next, progress }
}

File: flutter/lib/data/rewards.dart
/// Loyalty tiers + redeemable rewards. Mirrors data/rewards.ts.
class Reward {
  final String id;
  final String labelKey;
  final int pts;
  const Reward(this.id, this.labelKey, this.pts);
}

const rewards = <Reward>[
  Reward('items', 'rewards.r.items', 750),
  Reward('credit', 'rewards.r.credit', 2000),
  Reward('month', 'rewards.r.month', 6000),
];

class Tier {
  final String key;
  final int min;
  const Tier(this.key, this.min);
}

const tiers = <Tier>[
  Tier('Bronze', 0),
  Tier('Silver', 1500),
  Tier('Gold', 5000),
  Tier('Platinum', 12000),
];

Tier currentTier(int pts) {
  var t = tiers.first;
  for (final tier in tiers) {
    if (pts >= tier.min) t = tier;
  }
  return t;
}

Tier? nextTier(int pts) {
  for (final tier in tiers) {
    if (pts < tier.min) return tier;
  }
  return null;
}
